File: Back-End/src/main/java/com/nutrifacil/app/DTO/RegisterRequestDTO.java
package com.nutrifacil.app.DTO;

import com.nutrifacil.app.Enums.Gender;

import java.time.LocalDate;


public record RegisterRequestDTO(
        String username,
        String password,
        String fullname,
        String email,
        Gender gender,
        String birthdate,
        Double weight,
        Double height,
        DietDTO diet
        ) {
}

File: Front-End/site/src/types/RegisterRequestDTO.ts
export interface RegisterRequestDTO {
  username: string;
  password: string;
  fullname: string;
  email: string;
  gender: string;
  birthdate: Date;
  weight: number;
  height: number;
  allergies?: string[];
  diet: {
    objective: string,
    type: string,
    physicalActivityStatus: string,
    
  }
}
